Classes/Utility/Cache.php

```diff
@@ -83,6 +82,7 @@ public static function addCacheTagsByNewsRecords($newsRecords): void
     public static function addPageCacheTagsByDemandObject(NewsDemand $demand): void
     {
         $cacheTags = [];
+        $cacheDataCollector = $GLOBALS['TYPO3_REQUEST']->getAttribute('frontend.cache.collector');
```

Comment:
```suggestion
        $cacheDataCollector = self::getRequest()->getAttribute('frontend.cache.collector');
```
And add static method to get the request:
```
protected static function getRequest()
{
    return $GLOBALS['TYPO3_REQUEST'];
}
```